src/ApplicationInsights.Kubernetes/TelemetryInitializers/KubernetesTelemetryInitializer.cs

```diff
@@ -176,7 +177,14 @@ private void SetCustomDimension(ITelemetry telemetry, string key, string value)
 
             if (string.IsNullOrEmpty(value))
             {
-                _logger.LogError(Invariant($"Value is required to set custom dimension. Key: {key}"));
+                if (isValueOptional)
+                {
+                    _logger.LogTrace(Invariant($"Value is null or empty for key: {key}"));
```

Comment:
Do we even want to log this condition as a trace? It is *completely normal and expected* that a pod might not be created by a deployment, so there will be no deployment name. I am not sure if it helps with anything to have this in the logs.

Reply:
The trace level log reveals the code path, I don't see chatty becomes a problem. :-)